roles/heat/tasks/main.yml

```diff
@@ -53,21 +53,33 @@
 
 ## These heat openstack client calls can be replaced with ansible modules calls
 ## when we upgrade to ansible 2.0
+
 - name: create heat domain
   shell: . /root/stackrc; openstack --os-identity-api-version 3 --os-auth-url "{{ endpoints.keystone.url.admin }}/v3" domain create heat
   run_once: true
   register: heat_domain
   failed_when: heat_domain|failed and "HTTP 409" not in heat_domain.stderr
 
 - name: create heat domain admin user
-  shell: . /root/stackrc; openstack --os-identity-api-version 3 --os-auth-url "{{ endpoints.keystone.url.admin }}/v3" user create --domain heat --password "{{ secrets.stack_domain_admin_password }}" heat_domain_admin
+  os_user:
+    auth:
+      auth_url: "{{ endpoints.keystone.url.admin }}"
+      project_name: admin
+      username: admin
+      password: "{{ secrets.admin_password }}"
+    state: present
+    name: heat_domain_admin
+    password: "{{ secrets.stack_domain_admin_password }}"
+    domain: heat
   run_once: true
   register: heat_user
-  failed_when: heat_user|failed and "HTTP 409" not in heat_user.stderr
+
+- debug: var=heat_user
 
 - name: add admin role to the heat domain admin user
-  shell: . /root/stackrc; openstack --os-identity-api-version 3 --os-auth-url "{{ endpoints.keystone.url.admin }}/v3" role add --domain heat --user heat_domain_admin admin
+  shell: . /root/stackrc; openstack --os-identity-api-version 3 --os-auth-url "{{ endpoints.keystone.url.admin }}/v3" role add --domain heat --user "{{ heat_user.user.id }}" admin
   run_once: true
+  with_items: heat_user
```

Comment:
Previously we were registering the user above so that maybe we could create more than one?  but then I'd need to switch the code to {{ item }} so I'll remove it